scripts/sct_run_batch.py

```diff
@@ -332,8 +332,7 @@ def reset_streams():
     print('OS: ' + os_running + ' (' + platform.platform() + ')')
 
     # Display number of CPU cores
-    output = int(os.getenv('ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS', 0))
-    print('CPU cores: Available: {} | Used by SCT: {}'.format(multiprocessing.cpu_count(), output))
+    print('CPU cores: Available: {} | Threads used by SCT: {}'.format(multiprocessing.cpu_count(), args.itk_threads))
```

Comment:
I tried this PR with:

`sct_run_batch -jobs -1 -path-data ~/ds002902-download -path-output ~/ds002902-download_testing -continue-on-error 1 -script /usr/local/lib/spine-generic/process_data.sh`
 
It started computation on 15 cores (16 - 1), but in initial _INFO SYSTEM_ dialog, there is stated "Threads used by SCT: 1". Shouldn´t it be 15?

```
--
Spinal Cord Toolbox (git-cfh/run-batch-mpi_2806-aec57a3015cc3da4b00baa018b44cc95fb73f554)

INFO SYSTEM
-----------
OS: linux (Linux-5.4.0-0.bpo.2-amd64-x86_64-with-debian-10.5)
CPU cores: Available: 16 | Threads used by SCT: 1
RAM: Total 128606 MB | Available 124624 MB | Used 2661 MB
```

Reply:
Nope, passing `--itk-threads` should override `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS` (which it does slightly later in the code) which should limit the number of threads used by SCT to that number. This is the desired number right @jcohenadad?

Reply:
actually, `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS` only concerns the number of threads used for ITK-based functions (eg. ANTs), so I find it counter-intuitive to report "Threads used by SCT" with the variable `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS`. I would rather report the number of cores allocated for `sct_run_batch`, which in this case would be `--jobs`, right?

Reply:
Ahh, I took that as the intent from the original code which took this number as `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS` but before this env var was modified to match the `--itk-threads` argument. 

Given that the number of jobs is reported in the config, and that the number of threads used by non-ITK programs is unknowable, maybe we should just remove this from the output? 

Although to my knowledge we don't use threading anywhere in the python code, so threading would need to be coming from one of the other binaries.

Reply:
> Given that the number of jobs is reported in the config

yes, but i find it nice/reassuring to have it reported when the software runs. Esp. when ppl enter "-1" for jobs, they know for sure the number of cores that is allocated for parallel computing by `sct_run_batch` (does that make sense?).

Reply:
Makes sense, but we also don't do anything to stop the script from using more cores, so I think this should be reworded as `Running <n> parallel jobs` or something to that effect.

Reply:
ah, you're right. 

I was under the (wrong) impression we were, because when looking at the log file of one subject (ie. corresponds to one job), the output of `sct_check_dependencies` (which is ran within the script) gives:
~~~
Spinal Cord Toolbox (git-cfh/add-git-origin-in-sct-run-batch_2808-dd5fc48ad70a30df475e5fb6dc57b6f33e14a538)

SCT info:
- version: git-cfh/add-git-origin-in-sct-run-batch_2808-dd5fc48ad70a30df475e5fb6dc57b6f33e14a538
- path: /Users/julien/code/sct
OS: osx (Darwin-18.7.0-x86_64-i386-64bit)
CPU cores: Available: 16, Used by SCT: 1
~~~

So i thought that the multiprocessing "forced" to run the job with one core. **But**, after looking into `sct_check_dependencies`, the number of cores "Used by SCT" is actually the output of the variable `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS`, which is misleading (and often wrong!). So we should open an issue to address this, and report that the number of cores used by SCT is *not* limited, and specifically report the itk-thread variable (if useful, but it could also be confusing from the user perspective).

Reply:
here --> https://github.com/neuropoly/spinalcordtoolbox/issues/2849

Reply:
so, similarly for the suggestion in #2849, maybe we could also change the message in `sct_run_batch` for:
~~~
CPU cores: Available: 16 | Used by ITK functions: X
~~~

Reply:
I think, but I'm not sure, that multiprocessing doesn't implement core pinning, so `multiprocessing.Pool(4)` can launch four subprocesses, which can each go one to spawn jobs that use additional cores.

I'll open an issue for `sct_check_dependencies`

For `sct_run_batch` specifically, do we want to report both `--itk-threads` (`ITK_GLOBAL...` is set to this value for each job) and the computed number of jobs after resolving `0` and negative `-job` arguments?

Reply:
One additional point, I don't know if I trust all ITK programs to honour this environment variable, so even the `Threads used by ITK programs` may not be correct. Probably generally correct though.

Reply:
> yes, but i find it nice/reassuring to have it reported when the software runs. Esp. when ppl enter "-1" for jobs, they know for sure the number of cores that is allocated for parallel computing by sct_run_batch 

Sorry for spamming. I've added the ITK threads report back. Do we also want `Running n parallel` jobs in the output, for peace of mind that e.g. `-2` resolved to `14` parallel jobs?

Reply:
yes, but maybe not under the “system config” as the job number is more related to user’s input (although the same applies to itk thread which would be reported here). Not sure what’s best here. On the other hand job number is somewhat related to CPU cores...

Reply:
I added a new header after data called `RUNNING` where it prints the number of jobs running in parallel. Although I can almost see a reason to resolve everything (number of jobs, absolute paths, etc.) before printing the config. So even if you said `-2` jobs, the config will say `14` jobs. This reduces the dependency of the config on the system its run on, but ultimately I don't think it matters much.